jina/drivers/multimodal.py

```diff
@@ -0,0 +1,114 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from collections import defaultdict
+from typing import Iterable, Tuple, Dict
+
+import numpy as np
+
+from .encode import BaseEncodeDriver
+from .helper import pb2array, array2pb
+from ..proto import jina_pb2
+
+
+def _extract_doc_content(doc: 'jina_pb2.Document'):
+    """Returns the content of the document with the following priority:
+    If the document has an embedding, return it, otherwise return its content.
+    """
+    if doc.embedding.buffer:
+        return pb2array(doc.embedding)
+    else:
+        return doc.text or doc.buffer or (doc.blob and pb2array(doc.blob))
+
+
+def _extract_modalities_from_document(doc: 'jina_pb2.Document'):
+    """Returns a dictionary of document content (embedding, text, blob or buffer) with `modality` as its key
+    """
+    doc_content = {}
+    for chunk in doc.chunks:
+        modality = chunk.modality
+        if modality in doc_content:
+            return None
+        else:
+            doc_content[modality] = _extract_doc_content(chunk)
+    return doc_content
+
+
+class MultiModalDriver(BaseEncodeDriver):
```

Comment:
hmm, now i see `MultiModalDriver` is a grouped version of `BaseEncodeDriver`, the grouping happens at `extract_docs`